libyara/modules/pe.c

```diff
@@ -317,7 +321,7 @@ static void pe_parse_debug_directory(
       if (!struct_fits_in_pe(pe, pdb20, CV_INFO_PDB20))
         break;
       
-      pdb_str_len = strlen((char *)(pdb20->PdbFileName));
+      pdb_str_len = strnlen((char *)(pdb20->PdbFileName), MAX_PATH * sizeof(char));
```

Comment:
This is is an improvement over using `strlen` but it doesn't solve the original issue as you still can read out of the buffer's bounds. You are limiting the number of bytes being read to MAX_PATH, but the bytes remaining in the buffer can still be less than that. So, the argument you you pass to strnlen is the space available from the pdb20->PdbFileName pointer to the end of the buffer. You can use the `available_space` function for computing that.